packages/core/entity/FileDetail/index.ts

```diff
@@ -85,12 +87,69 @@ export default class FileDetail {
         return FileDetail.convertAicsS3PathToHttpUrl(`${AICS_FMS_S3_BUCKET}${pathWithoutDrive}`);
     }
 
+    private static generateFilePath(env: Environment, fileName: string, fmsId: string): string {
+        if (!fileName || !fmsId) {
+            throw new Error("File name and FMS ID must be provided");
+        }
+
+        // Define prefixes based on the environment
+        const prefixes: Record<Environment, string> = {
+            LOCALHOST: "/tmp/fss/local",
+            PRODUCTION: "/allen/programs/allencell/data/proj0",
+            STAGING: "/allen/aics/software/apps/staging/fss/data",
+            TEST: "test",
+        };
+
+        // Get the prefix for the given environment
+        const prefix = prefixes[env];
+        if (!prefix) {
+            throw new Error(`Invalid environment: ${env}`);
+        }
+
+        // Generate path segments from FMS ID
+        const pathSegments = this.convertFMSIDToLocalPath(fmsId);
+
+        // Construct the full file path
+        const filePath = `${prefix}/${pathSegments.join("/")}/${fileName}`;
+
+        return filePath;
+    }
+
+    private static convertFMSIDToLocalPath(guid: string): string[] {
+        if (!guid) {
+            throw new Error("GUID cannot be null or undefined");
+        }
+
+        const paths: string[] = [];
+
+        while (guid.length > 2) {
```

Comment:
This is great, but FWIW lodash can do this whole method in one line for ya:
```

    // Splits the fileId into an array where is character is its own element then chunks them into groups of 3 with the
    // remainder (between 1-3 characters) becoming the final group
    chunk(this.fileId.split(""), 3);
```

Reply:
oh yeah I definitely just ported it over from FSS and didnt consider.

Reply:
less custom code to test FTW

Reply:
Resolved by deriving from cloud path in [3bee712](https://github.com/AllenInstitute/biofile-finder/pull/389/commits/3bee712c637ad2bf7571bed19b960fe11de85598)